third_party/nlohmann_json/CMakeLists.txt

```diff
@@ -6,5 +6,6 @@ FetchContent_Declare(
     GIT_TAG        v3.9.1
 )
 set(JSON_BuildTests OFF CACHE INTERNAL "")
-set(JSON_Install OFF CACHE INTERNAL "")
+set(JSON_Install ON CACHE INTERNAL "")
```

Comment:
I agree it will mess something up. but if they want but do not have the required library, I think we need to install the dependency. Otherwise, the ginkgo installation will depend on something from build folder not the installation place

Reply:
This is a blocker to me (see [How to make package managers cry, slide 13](https://archive.fosdem.org/2018/schedule/event/how_to_make_package_managers_cry/attachments/slides/2297/export/events/attachments/how_to_make_package_managers_cry/slides/2297/how_to_make_package_managers_cry.pdf)), and a reason to justify making the public `json_parser` header either conditional only on the presence of an external nlohmann-json installation or not conditional at all (excluding it from `ginkgo.hpp`). The former makes it hard to impossible to test the JSON parser, which is why I suggest the latter.
The installation does not depend on anything in the linking sense then, only in the interface sense, which is perfectly fine for an isolated component related to a header-only library IMO.